### PR TITLE
AZP: Remove io_demo tests for CX-4

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
## What
Remove io_demo tests for CX4.

## Why ?
CX-4 EOL.

### CODE REVIEW

File: buildlib/pr/io_demo/io-demo.yml [hunk 1]
```diff
@@ -81,20 +77,15 @@ jobs:
...
           interface: $(roce_iface_cx6)
           tls: "dc_x"
           test_name: am_umemh_cx6_dc
-        "Multi HCAs: tag match on CX4 and CX6 RC ":
-          interface: $(roce_iface_cx4) $(roce_iface_cx6)
+        "Multi HCAs: tag match CX6 RC ":
+          interface: $(roce_iface_cx6)
           tls: "rc_x"
-          test_name: multy_cx6_cx4_rc
+          test_name: multy_cx6_rc
```
Comment (yosefe) [inline buildlib/pr/io_demo/io-demo.yml:88]: seems this test is no longer needed, it's not multi-HCA any more
let's add testing over BF3 instead. could be in next PR.

Comment (dpressle) [reply buildlib/pr/io_demo/io-demo.yml:88]: Yes, let do it in a separate PR

**Review by dpressle — APPROVED** [review]

**Review by yosefe — COMMENTED** [review]
can you pls remove the multi test as well in this PR?

**Review by brminich — APPROVED** [review]

**Comment by Alexey-Rivkin** [comment]
@yosefe can I squash?

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**